osu.Game/Overlays/BeatmapSet/BeatmapRulesetSelector.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Events;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class BeatmapRulesetSelector : RulesetSelector
+    {
+        private readonly Box bar;
+
+        private BeatmapSetInfo beatmapSet;
+
+        public BeatmapSetInfo BeatmapSet
+        {
+            get => beatmapSet;
+            set
+            {
+                beatmapSet = value;
+
+                this.FadeTo(beatmapSet != null ? 1 : 0.25f);
+
+                foreach (var tabItem in TabContainer)
+                    ((RulesetTabItem)tabItem).BeatmapsCount = value?.Beatmaps.FindAll(r => r.Ruleset.ID == tabItem.Value.ID).Count ?? 0;
+            }
+        }
+
+        protected override TabItem<RulesetInfo> CreateTabItem(RulesetInfo value) => new RulesetTabItem(value)
+        {
+            Anchor = TabContainer.Anchor,
+            Origin = TabContainer.Origin,
+        };
+
+        public BeatmapRulesetSelector()
+        {
+            TabContainer.Anchor = Anchor.BottomCentre;
+            TabContainer.Origin = Anchor.BottomCentre;
+            TabContainer.Spacing = new Vector2(15, 0);
+
+            AddInternal(bar = new Box
+            {
+                Anchor = TabContainer.Anchor,
+                Origin = TabContainer.Origin,
+                RelativeSizeAxes = Axes.X,
+                Height = 1,
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            bar.Colour = colours.Yellow;
+        }
+
+        /// <summary>
+        ///  Selects a <see cref="RulesetTabItem"/> that contains a matching <see cref="RulesetInfo"/>.
+        /// </summary>
+        public void SelectRuleset(RulesetInfo ruleset)
+        {
+            if (ruleset == null)
+                return;
+
+            foreach (var tab in TabContainer)
+                if (tab.Value.ID == ruleset.ID)
+                    SelectTab(tab);
+        }
+
+        private class RulesetTabItem : TabItem<RulesetInfo>
+        {
+            private readonly OsuSpriteText text;
+            private readonly Box hoverBox;
+
+            private readonly OsuSpriteText count;
+            private readonly Container countContainer;
+
+            private OsuColour colours;
+
+            public int BeatmapsCount
+            {
+                set
+                {
+                    Enabled.Value = value > 0;
```

Comment:
This won't correctly handle the initial state. I suggest moving all of this to a bindable:

```
Bindable<int> BeatmapCount;

...

LoadComplete():
    BeatmapCount.BindValueChanged(..., true);
```